js/views/TestProfilePage.js

```diff
@@ -45,8 +69,15 @@ export default class extends BaseVw {
     social.remove(social.at(index));
   }
 
+  onRemoveSocial() {
+    // since the validation is index based and
+    // removing alters the indexes in the collection,
+    // we need to revalidate
```

Comment:
Yeah, the current approach with validation of a nested collection I took involves keeping indexes in the validation error keyname, e.g:

```
{
  social[0].type: ['an error'],
  social[1].type: ['an error', 'another error'],
}
```

Similarly, I'm using an index approach for the field name in the DOM. This is used when we scrape the form in order to extract the data from it to put in the model (similar to the JSON form plugin you showed me), e.g:

```
<div class="socialAccount">
  <input type="text" name="social[0].type" value="boom" />
</div>

<div class="socialAccount">
  <input type="text" name="social[1].type" value="bam" />
</div>
```

This ends up being problematic for a couple of reasons:

1.) Let's say you use jQuery to remove / add new items. This will leave the indexes out of sync and when you scrape the DOM, you end up with structure likes this (assuming you removed the first account above):

```
{[undefined, { type: "bam"}]}
```

2.) # 2 is similar to # 1, but with validation. Any time you remove an item from the collection, you'll need to re-validate the model because any errors in the `validationError` object may have incorrect indexes.

Anyhow, I think an approach where we could fix this issue is, instead of relying on index (which is something that changes for a model within a collection), we could use the model's cid as a way to map between the model and it's error / DOM field.

But... I need to let this feature breathe for a bit and then I'll circle back.